### PR TITLE
Update solidity

### LINKED ISSUE #10: Fix bug where approvers count goes up even when the person has already contributed

### LINKED ISSUE #11: Upgrade Solidity version
We are currently using an old version of Solidity and the solc compiler. Updating the compiler will cause the contracts to fail because the new solc compiler doesn't support the Solidity version we are using.

### PR DESCRIPTION
**Pull request by jpc20**
# Pull request checklist

## Pull request type

Please check the type of change your PR introduces:

- [x] Bugfix
- [ ] Feature
- [x] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## Current behavior

- Our Solidity version and compiler version were both deprecated. We were also getting errors for exporting a mutable value from `web3.js` and had a bug which caused the approverCount to go up even if the person donating had already contributed.

## New behavior

- Solidity and compiler versions have been updated. This required changes to the contract code because the update broke our version of the code.
- Approver count for a campaign now only goes up once per donor.
- `web3.js` is now exporting a `const`

## Other information

closes #11 
closes #10

### CODE REVIEW

File: ethereum/contracts/Campaign.sol [hunk 1]
```diff
@@ -29,21 +30,21 @@ contract Campaign {
     function contribute() public payable {
         require(msg.value > minimumContribution);
 
-        approvers[msg.sender] = true;
-        approversCount++;
+        if (approvers[msg.sender] != true) {
+          approvers[msg.sender] = true;
+          approversCount++;
+        }
```
Comment (jpc20) [inline ethereum/contracts/Campaign.sol:36]: This fixes the bug that caused the approvers count to go up even when the person has already donated.

File: ethereum/contracts/Campaign.sol [hunk 2]
```diff
@@ -29,21 +30,21 @@ contract Campaign {
...
-        requests.push(newRequest);
+    function createRequest(string memory description, uint value, address payable recipient)
+        public restricted returns (uint requestID){
+          requestID = requestsCount++;
+          Request storage newRequest = requests[requestID];
+          newRequest.description = description;
+          newRequest.value = value;
+          newRequest.recipient = recipient;
+          newRequest.complete = false;
+          newRequest.approvalCount = 0;
```
Comment (jpc20) [inline ethereum/contracts/Campaign.sol:47]: This was the biggest thing that broke when the update happened. I'm not totally sure what happened, but I was able to find this type of set up in the Solidity Docs and it is working now.

File: ethereum/web3.js [hunk 3]
```diff
@@ -1,15 +1,9 @@
...
-// eslint-disable-next-line import/no-mutable-exports
-let web3;
-
-if (typeof window !== "undefined" && typeof window.web3 !== "undefined") {
-  web3 = new Web3(window.web3.currentProvider);
-} else {
-  const provider = new Web3.providers.HttpProvider(
+const web3 = (typeof window !== "undefined" && typeof window.web3 !== "undefined")
+  ? new Web3(window.web3.currentProvider)
+  : new Web3(new Web3.providers.HttpProvider(
```
Comment (jpc20) [inline ethereum/web3.js:5]: This was giving me an error for exporting a `let` so I switched it to `const`

File: test/Campaign.test.js [hunk 4]
```diff
@@ -82,4 +83,17 @@ contract("Campaign", async accounts => {
...
+      let campaign = await Campaign.deployed();
+      await campaign.contribute({from: accounts[1], value: '200'});
+      let summary = await campaign.getSummary()
+      let approversCount = summary[3].toString();
+
+      assert.equal(1, approversCount);
+
+      await campaign.contribute({from: accounts[1], value: '200'});
+
+      assert.equal(1, approversCount);
```
Comment (jpc20) [inline test/Campaign.test.js:97]: Testing that approversCount only goes up once per account that donates.

**Review by attom2 — APPROVED** [review]
Looks good, glad you got the bug figured out

**Pull request merged**